.github/workflows/email-check.yaml

```diff
@@ -0,0 +1,40 @@
+name: "Check for private emails used in PRs"
+
+on: pull_request_target
+
+permissions:
+  contents: read
+
+jobs:
+  validate_email:
+    permissions:
+      pull-requests: write
+    runs-on: ubuntu-latest
+    if: github.repository == 'llvm/llvm-project'
+    steps:
+      - name: Fetch LLVM sources
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
```

Comment:
Is this getting the whole repo each time? I wonder if there are some "shallow checkout" options that would apply here.

Or we could get it from the PR event, maybe, but I guess you already looked for that and I don't want to derail the chosen solution.

Reply:
There is `user.email` in https://docs.github.com/en/webhooks/webhook-events-and-payloads#pull_request. Would need testing to confirm, but I'd guess that checking for null or `noreply@github...` in the string would work.

And possibly this could all be done in the YAML. Which is not necessarily a good thing so you don't have to change what you've got.

Just know that this is (allegedly) an option.

Reply:
Emails could come from different places, but we mostly bother about email of author in the commit (which has to set up manually from via `git config` if one uses git CLI and private email address, so must be done explicitly) and I thought this is the most bullet-proof way to do this.